connect/api/src/main/java/org/apache/kafka/connect/storage/Converter.java

```diff
@@ -44,11 +45,34 @@
      */
     byte[] fromConnectData(String topic, Schema schema, Object value);
 
+    /**
+     * Convert a Kafka Connect data object to a native object for serialization.
```

Comment:
Might be worthwhile to add a bit more here about which are now used and expected to be implemented. For example, maybe something along these lines in the JavaDoc of both new methods?
```
* Convert a Kafka Connect data object to a native object for serialization,
* potentially using the supplied topic and headers in the record as necessary.
*
* <p>Connect uses this method directly, and for backward compatibility reasons this method
* by default will call the {@link #fromConnectData(String, Schema, Object)} method.
* Override this method to make use of the supplied headers.
*
```

Reply:
I like your wording, added